tests/test_indexes.py

```diff
@@ -191,18 +191,25 @@ def test_range_index_union():
 
 
 def test_interval_range():
-    check(assert_type(pd.interval_range(0, 10), pd.IntervalIndex), pd.IntervalIndex)
+    check(
+        assert_type(pd.interval_range(0, 10), "pd.IntervalIndex[pd.Interval[int]]"),
+        pd.IntervalIndex,
```

Comment:
could add `int` as last argument to `check` (similar for other tests that return `IntervalIndex`), with appropriate types

Reply:
They are actually `pd.Interval` not `int`, so it is a bit pointless since this is the only type IntervalIndex can have.